src/Native/Unix/System.Security.Cryptography.Native/pal_dsa.c

```diff
@@ -21,7 +21,7 @@ void CryptoNative_DsaDestroy(DSA* dsa)
 int32_t CryptoNative_DsaGenerateKey(DSA** dsa, int32_t bits)
 {
     *dsa = DSA_new();
-    if (!dsa)
+    if (!*dsa)
```

Comment:
I think a better fix is to move the DSA_new to after this.  It's making sure that we're not about to SIGSEGV.

Reply:
Thanks, that makes sense. Should this code also try and check the value returned by `DSA_new`  too?

Reply:
I guess it makes sense to also check, since it's about to dereference it, and it's documented as being capable of returning null (but that shouldn't assert, since it's an API condition)

Reply:
Fixed now